### ISSUE TITLE
Add opinion citations  to every RECAP document

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#607]
This is a big one. The text of every citation in every RECAP document should be linkified to the opinion that it is referring to. In addition:

 - There should be a new Table of Authorities page for every document, just like for opinions. [Here's an example][ex]. There should also be a matching sidebar for every RECAP document listing the first ten items that it links to (probably listed by relevancy).

 - You should be able to search the RECAP content by citation, [like you can with opinions][citation-search]. On opinions, there could be a new button that says, "See PACER documents citing this opinion." That link would take you to a recap search for something like: `cites:22394`. 

 - Eventually, this could probably also be used as a factor for citegeist.

[citation-search]: https://free.law/2016/01/30/citation-searching-on-courtlistener/
[ex]: https://www.courtlistener.com/opinion/108713/roe-v-wade/authorities/

**Comment by ss108** [comment]
Parentheticals cannot be stored currently for RECAP documents because the `describing_opinion` field is a foreign key to `Opinion`.

https://github.com/freelawproject/courtlistener/blob/main/cl/search/models.py#L3097

One way to handle this in the future is to: 

1) add a column indicating the type of the document where the parenthetical exists (like an enum or something)
2) add another column `describing_document` that can point to either a `RECAPDocument` or `Opinion`
3) run a batch job to update existing entries in the table
4) stop writing to the old column, update everything to write to the new, etc
[5) delete the old column after an interval]

**Comment by mlissner** [comment]
FWIW, I think parentheticals in the RECAP corpus probably are not as valuable because they are not guaranteed to be written by a judge. One of the nice things about parentheticals in opinions is that you can be certain¹ they're written by somebody that knows what they're talking about. That's not true for all the filings in RECAP.


<sub>1. Well, you know the person has a law degree and became a judge. That's *something* at least. </sub>

**Comment by ss108** [comment]
I had a code comment that you wanted extracted for future discussion: ```    
    """
    TODO: check whether need to show cited opinions on front-end in RECAP doc pages
    """``` 

I think for now, having a TOA section for every document is sufficient, but I do think that, in the future, we should eschew the plain text section for a docket entry and make it a fancier "Courtlistener native" page that has, inter alia, the citation links in-lined the way they are for Opinions.

**Comment by mlissner** [comment]
Yeah, I think adding a tab to each document page for the TOA would be useful (see opinions for prior art), and that having a TOA tab for the case would be really neat too, particularly if we can think of a simple way to rank the citations.

**Comment by ss108** [comment]
Just making a note of this as well:

> Update the referenced citation with a count of inbound citations

**Comment by mlissner** [comment]
Thrilled to close this seven years after it was first created, and about eight months after you started working on it, @ss108. Thank you again!
